File: src/main/java/axirassa/ioc/IocUnitTestRunner.java
package axirassa.ioc;

import org.apache.tapestry5.ioc.Registry;
import org.apache.tapestry5.ioc.RegistryBuilder;
import org.junit.runners.BlockJUnit4ClassRunner;
import org.junit.runners.model.InitializationError;
import org.mockito.MockitoAnnotations;

/**
 *
 * @author wiktor
 */
public class IocUnitTestRunner extends BlockJUnit4ClassRunner {

    private final Registry registry;

    public IocUnitTestRunner (Class<?> classObject) throws InitializationError {
        super(classObject);

        RegistryBuilder builder = IocStandalone.initRegistryBuilder();
        builder.add(MockingModule.class);

        registry = builder.build();
        registry.performRegistryStartup();
    }


    @Override
    public Object createTest () {
        Object test = registry.autobuild(getTestClass().getJavaClass());
        MockitoAnnotations.initMocks(test);

        return test;
    }


}

File: src/main/java/axirassa/ioc/IocUnitTestRunner.scala
package axirassa.ioc

import org.apache.tapestry5.ioc.Registry
import org.junit.runners.BlockJUnit4ClassRunner
import org.mockito.MockitoAnnotations

/**
 *
 * @author wiktor
 */
class IocUnitTestRunner(classObject:Class[_]) extends BlockJUnit4ClassRunner(classObject) {

    private var registry: Registry = _

    val builder = IocStandalone.initRegistryBuilder()
    builder.add(classOf[MockingModule])

    registry = builder.build()
    registry.performRegistryStartup()


    override
    def createTest () = {
        val test = registry.autobuild(getTestClass().getJavaClass())
        MockitoAnnotations.initMocks(test)

        test.asInstanceOf[Object]
    }


}
